### PR TITLE
feat(asset-api): use the asset API for exporting assets

### PR DESCRIPTION
**Pull request by obs-gh-colinhutchinson**
This PR amounts to a rewrite.

Instead of using the GCP API we're going to use the [GCP Asset Inventory API](https://cloud.google.com/asset-inventory/docs/overview) to export asset information to a GCP Bucket then from the Bucket parse the information into the pub/sub

related: https://github.com/observeinc/terraform-google-collection/pull/32

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -1,338 +1,256 @@
...
+    bucket = storage_client.get_bucket(data["bucket"])
+
+    blob = bucket.get_blob(data["name"])
+
+    # Check if blob is None
+    if blob is None:
+        logging.warning("Blob is None, returning without further action")
+        return
+
+    # Skip processing if filename starts with "temp"
```
Comment (bendrucker) [inline main.py:195]: This comment doesn't add any information (same with the one for None FWIW). As I'm reviewing this, I'm wondering _why_ these conditions would occur. The `if` statements and early return are sufficiently are sufficiently clear about what's happening.

The log messages are also a bit too closely coupled to the code. Messages that would convey more, for example:

* `'Blob %q not found, skipping' % name`
* `'Blob %q is temporary, skipping' % name`

If my interpretations are wrong, all the more reason to have meaningful log messages, so I don't have to guess what they mean in practice.

File: main.py [hunk 2]
```diff
@@ -1,338 +1,256 @@
...
+    except json.JSONDecodeError as e:
+        return
+
+    # Extract content_type and asset_type from the GCS bucket path
+    gcs_prefix = f"{bucket.name}/"
+    path = data["name"][len(gcs_prefix):]
+    folders = path.split("/")
+    if len(folders) < 4:
+        logging.warning(
+            "Path structure unexpected, returning without further action")
```
Comment (bendrucker) [inline main.py:220]: Also unclear, something like `"expected $foo/$content_type/$asset_type, got %q" % path`

File: .dockerignore [hunk 3]
```diff
@@ -0,0 +1,89 @@
+# Git
+.git
+.gitignore
+.gitattributes
+
+
+# CI
```
Comment (bendrucker) [inline .dockerignore:7]: Avoid boilerplate everything-that-might-exist ignore files like this. Ignore the things you actually have in the project and want to skip.

**Review by bendrucker — APPROVED** [review]
Approved on the main substance, ie the Python bits, but I'd really like to see the whole development/Docker situation addressed. When you show up to this (public) repo it now seems like this application is packaged with Docker for production use. And after a full review that seems to not be the case.

File: entrypoint.sh [hunk 4]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+set -eo pipefail
+
+if [ -n "${DEBUG:-}" ]; then
+    set -x
+fi
+
+if [ "${ENV}" == "dev" ]; then
```
[the remark this answers is not in the document]
Comment (obs-gh-colinhutchinson) [reply entrypoint.sh:8]: Initially was only using it for development but expanded that today to include the ability to run the unittests

**Pull request merged**